### ISSUE TITLE
Pay for Order page: currency switching doesn't work

### ISSUE THREAD

**Issue by jacoswan** [Automattic/woocommerce-payments#5031]
## Describe the bug

Currency switching doesn't work on the Pay for Order page and displays store base currency regardless. 

Doesn't matter what the base currency is and whether or not the _New Payment Exprience_ is enabled. 

## To Reproduce

1. Create an order for a customer
2. Visit the Customer Payment Page
3. The store's base currency is displayed. 
4. Attempts to change it fail

## Actual behavior

Pay for Order page's currency should update as the rest of the shop should, based on settings. 

## Screenshots

Even though the store says it knows the visitor is in the UK and that the currency will be updated, it isn't: 

![2zofBK.png](https://user-images.githubusercontent.com/15178758/199195536-17212ef5-e35c-4920-9118-14abef7aea5f.png)

The Pay for Order page doesn't respond, but the rest of the store does:

![5e13lS.gif](https://user-images.githubusercontent.com/15178758/199195830-ab93d8b3-51c3-4719-aa7c-6e65ba07d400.gif)

## Expected behavior

Pay for Order page should work the same as the rest of the store does. 

### Desktop (please complete the following information):

* OS: MacOS Big Sur 11.6.1 (20G224)
* Browser: Chrome
* Version: 106.0.5249.119 (Official Build) (x86_64)

### Additional context

5665064-zen

`
### WordPress Environment ###

WordPress address (URL): https://woo.test
Site address (URL): https://woo.tst
WC Version: 7.0.0
REST API Version: ✔ 7.0.0
WC Blocks Version: ✔ 8.5.1
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0.3
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.1.11
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.85.0
OpenSSL/1.1.1n

SUHOSIN Installed: –
MySQL Version: 10.4.26-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.0.0
WC Database Prefix: wp_
Total Database Size: 17.59MB
Database Data Size: 12.64MB
Database Index Size: 4.95MB
wp_woocommerce_sessions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.17MB + Index: 0.16MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 1.52MB + Index: 0.45MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.28MB + Index: 0.19MB + Engine InnoDB
wp_advanced_notifications: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_advanced_notification_triggers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_automatewoo_abandoned_carts: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_automatewoo_customers: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_automatewoo_customer_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_events: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_guests: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_automatewoo_guest_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_log_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_queue: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_queue_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.14MB + Index: 0.09MB + Engine InnoDB
wp_FmeCCFA_fields: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_followup_coupons: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_coupon_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_followup_customers: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_followup_customer_carts: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_customer_notes: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_customer_orders: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_followup_email_excludes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_followup_email_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_followup_email_orders: Data: 0.13MB + Index: 0.09MB + Engine InnoDB
wp_followup_email_tracking: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_followup_followup_history: Data: 0.06MB + Index: 0.05MB + Engine InnoDB
wp_followup_order_categories: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_order_items: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_followup_subscribers: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_subscribers_to_lists: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_followup_subscriber_lists: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_gla_budget_recommendations: Data: 0.20MB + Index: 0.11MB + Engine InnoDB
wp_gla_merchant_issues: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wp_gla_shipping_rates: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_gla_shipping_times: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_gp_glossaries: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_gp_glossary_entries: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_gp_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_gp_originals: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_gp_permissions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_gp_projects: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_gp_translations: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_gp_translation_sets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mollie_pending_payment: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_ms_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_nf3_actions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_nf3_action_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_nf3_chunks: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_nf3_fields: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_nf3_field_meta: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_nf3_forms: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_nf3_form_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_nf3_objects: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_nf3_object_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_nf3_relationships: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_nf3_upgrades: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 4.08MB + Index: 0.14MB + Engine InnoDB
wp_postmeta: Data: 1.52MB + Index: 0.64MB + Engine InnoDB
wp_posts: Data: 0.23MB + Index: 0.14MB + Engine InnoDB
wp_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_termmeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.06MB + Index: 0.05MB + Engine InnoDB
wp_term_taxonomy: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.08MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wcpv_commissions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcpv_per_product_shipping_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcs_payment_retries: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availability: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availabilitymeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_booking_relationships: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_deposits_payment_plans: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_deposits_payment_plans_schedule: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_gpf_render_cache: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.05MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_exported_csv_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_gpf_google_taxonomy: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_table_rates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_square_customers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wp_phpmyadmin_extension__errors_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_indexable: Data: 0.05MB + Index: 0.08MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_yoast_seo_meta: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_yoco_order_errors: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

attachment: 150
aw_workflow: 5
bookable_resource: 4
custom_css: 1
event_ticket: 4
follow_up_email: 1
nav_menu_item: 9
odr_master_log: 1
odr_order_bump: 1
odr_template: 1
oembed_cache: 4
page: 20
post: 2
product: 83
product_variation: 29
revision: 51
shop_coupon: 4
shop_order: 132
shop_subscription: 8
wc_booking: 17
wc_membership_plan: 2
wc_order_status: 8
wc_user_membership: 2
wcpf_item: 1
wcpf_project: 1
wp_global_styles: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (3) ###

Jetpack: by Automattic – 11.5-beta4
WooCommerce Payments: by Automattic – 4.9.0
WooCommerce: by Automattic – 7.0.0

### Inactive Plugins (11) ###

Akismet Anti-Spam: by Automattic – 5.0.1
Bookings Helper: by WooCommerce – 1.0.3
Google Listings and Ads: by WooCommerce – 2.2.0
Loco Translate: by Tim Whitlock – 2.6.2
WooCommerce Bookings: by WooCommerce – 1.15.62 (update to version 1.15.64 is available)
WooCommerce GoCardless Gateway: by WooCommerce – 2.5.1 (update to version 2.5.2 is available)
WooCommerce Order Barcodes: by WooCommerce – 1.5.2
WooCommerce Shipping & Tax: by WooCommerce – 1.26.3
WooCommerce Square: by WooCommerce – 3.2.0
WooCommerce Stripe Gateway: by WooCommerce – 6.9.0
WooCommerce Subscriptions: by WooCommerce – 4.6.0

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Settings ###

API Enabled: –
Force SSL: –
Currency: AUD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: booking (booking)
composite (composite)
external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude_from_catalog (exclude_from_catalog)
exclude_from_search (exclude_from_search)
exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔
Enforce Approved Product Download Directories: –

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.1.3
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Subscriptions ###

WCS_DEBUG: ✔ No
Subscriptions Mode: ✔ Live
Subscriptions Live URL: https://woo.tst
Subscription Statuses: wc-cancelled: 8
WooCommerce Account Connected: ✔ Yes
Active Product Key: ❌ No

### Store Setup ###

Country / State: Australia — New South Wales

### Subscriptions by Payment Gateway ###

other: wc-cancelled: 2
WooCommerce Payments: wc-cancelled: 1

### Payment Gateway Support ###

WooCommerce Payments: products
refunds
multiple_subscriptions
subscription_cancellation
subscription_payment_method_change_admin
subscription_payment_method_change_customer
subscription_payment_method_change
subscription_reactivation
subscription_suspension
subscriptions
gateway_scheduled_payments
tokenization
add_payment_method

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
experimental-product-tour
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
multichannel-marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
settings

Daily Cron: ✔ Next scheduled: 2022-11-02 04:18:59 -04:00
Options: ✔
Notes: 114
Onboarding: completed

### WooCommerce Payments ###

Version: 4.9.0
Connected to WPCOM: Yes
Blog ID: 167752461
Account ID: acct_1Jysip2H9KWIXo0t

### Action Scheduler ###

Complete: 1,334
Oldest: 2022-10-01 07:24:21 -0400
Newest: 2022-11-01 04:55:41 -0400

Failed: 6
Oldest: 2020-03-26 01:54:28 -0400
Newest: 2022-05-09 08:45:07 -0400

Pending: 11
Oldest: 2022-11-01 05:00:41 -0400
Newest: 2023-04-20 00:00:00 -0400

### Status report information ###

Generated at: 2022-11-01 04:59:12 -04:00
`

**Comment by csmcneill** [comment]
I opened a duplicate issue in #4964 but it was closed due to the nature of how multi-currency works. @jessepearson said the following:

> The currency and the exchange rate are set in the order meta at the time the order is created, and that becomes the currency for the order from that point forward. If the merchant needs to create an order for a customer in a different currency, they would need to place the order through the front end of the site.

There are a couple of workarounds in [this comment](https://github.com/Automattic/woocommerce-payments/issues/4964#issuecomment-1294832456).

**Comment by jessepearson** [comment]
I think there's a couple things that can be addressed in this issue:

* We can see if we can hide the widget itself if the page is the "pay for order" page. I thought we had done this in the past, but it may not have been done.
* We can make sure the automatic banner doesn't show on the "pay for order" page, as well. 

If those two things are done, it would help at least reduce the confusion for both the customer and merchant.

**Comment by jessepearson** [comment]
I came across this again when looking at another issue and quickly looked into why the widgets were showing on the `pay_for_order` page. It appears that nothing was added to stop the widgets from showing, coupled with the banner stating the currency was changed when it actually wasn't means this was a bug and not an enhancement, so I was wrong for changing this previously.

I've created a PR to fix both issues, which is linked above.
